python/semantic_kernel/processes/kernel_process/kernel_process.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from typing import TYPE_CHECKING, Any
+
+from pydantic import Field
+
+from semantic_kernel.processes.kernel_process.kernel_process_edge import KernelProcessEdge
+from semantic_kernel.processes.kernel_process.kernel_process_state import KernelProcessState
+from semantic_kernel.processes.kernel_process.kernel_process_step_info import KernelProcessStepInfo
+from semantic_kernel.processes.local_runtime.local_event import KernelProcessEvent
+from semantic_kernel.processes.local_runtime.local_kernel_process_context import LocalKernelProcessContext
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+if TYPE_CHECKING:
+    from semantic_kernel.kernel import Kernel
+    from semantic_kernel.processes.kernel_process.kernel_process_edge import KernelProcessEdge
+
+
+@experimental_class
+class KernelProcess(KernelProcessStepInfo):
+    """A kernel process."""
+
+    steps: list[KernelProcessStepInfo] = Field(default_factory=list)
+
+    def __init__(
+        self,
+        state: KernelProcessState,
+        steps: list[KernelProcessStepInfo],
+        edges: dict[str, list["KernelProcessEdge"]] | None = None,
+    ):
+        """Initialize the kernel process."""
+        if not state:
+            raise ValueError("state cannot be None")
+        if not steps:
+            raise ValueError("steps cannot be None")
+        if not state.name:
+            raise ValueError("state.Name cannot be None")
+
+        process_steps = []
+        process_steps.extend(steps)
+
+        args: dict[str, Any] = {
+            "steps": process_steps,
+            "inner_step_type": KernelProcess,
+            "state": state,
+            "output_edges": edges or {},
+        }
+
+        super().__init__(**args)
+
+    async def start(self, kernel: "Kernel", initial_event: KernelProcessEvent) -> LocalKernelProcessContext:
+        """Start the kernel process."""
+        async with LocalKernelProcessContext(self, kernel) as process_context:
+            await process_context.start_with_event(initial_event)
+            return process_context
```

Comment:
Is the context still valid after leaving this block?

Reply:
Great question! I was just looking into this more. I was hoping to control the resource clean up by using the `async with` in that way, but your question causes me to think more about it. It would probably be better to have the caller responsible for managing the resources and thus they could do:

```python
    async with await kernel_process.start(
        kernel=kernel, initial_event=CommonEvents.StartProcess, data="foo"
    ) as process_context:
        process_state = await process_context.get_state()
        c_step_state: KernelProcessStepState[CStepState] = next(
            (s.state for s in process_state.steps if s.state.name == "CStep"), None
        )
        assert c_step_state.state  # nosec
        assert c_step_state.state.current_cycle == 3  # nosec
        print(f"CStepState current cycle: {c_step_state.state.current_cycle}")
```